### PR TITLE
8259511: java/awt/Window/MainKeyWindowTest/TestMainKeyWindow.java failed with "RuntimeException: Test failed: 20 failure(s)"

### PR DESCRIPTION
**Pull request by mrserb**
This test failed on the system where the wrong resolution was used(too small). Looks like that resolution was set by one of the tests and was not reset back because of a misconfigured automatic system reboot. I have checked that the initial version of the test works fine on our "clean" systems. But would like to tweak the test anyway just in case.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8259511](https://bugs.openjdk.java.net/browse/JDK-8259511): java/awt/Window/MainKeyWindowTest/TestMainKeyWindow.java failed with "RuntimeException: Test failed: 20 failure(s)"

### Reviewers
 * [Jayathirth D V](https://openjdk.java.net/census#jdv) (@jayathirthrao - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2113/head:pull/2113`
`$ git checkout pull/2113`

### CODE REVIEW

File: test/jdk/java/awt/Window/MainKeyWindowTest/TestMainKeyWindow.java [hunk 1]
```diff
@@ -388,7 +388,7 @@ public static void main(String[] args) throws Exception
             MenuBar mb = new MenuBar();
             mb.add(new Menu("Hello"));
             frame.setMenuBar(mb);
-            frame.setBounds(400, 180, 1068, 821);
+            frame.setBounds(400, 180, 300, 300);
```
Comment (mrserb) [inline test/jdk/java/awt/Window/MainKeyWindowTest/TestMainKeyWindow.java:391]: When a small resolution(800x600) or similar is used, then the window manager moves this big window to the left/top but the test is expected the specific x/y coordinate.

**Comment by jayathirthrao** [comment]
Hi Sergey,

I am trying to verify the fix in my mac, if i simply run the test without any change it is throwing error : "Use -nativepath to specify the location of native code".
Do we need to change anything in the mac preferences to run this test?

Thanks,
Jay

**Comment by mrserb** [comment]
> I am trying to verify the fix in my mac, if i simply run the test without any change it is throwing error : "Use -nativepath to specify the location of native code".
> Do we need to change anything in the mac preferences to run this test?

This test has a native part so you need to specify this option and point it to the built test bundle, or you can run it via make file:
`make test TEST="jtreg:open/test/jdk/java/awt/Window/MainKeyWindowTest/TestMainKeyWindow.java"`

**Comment by jayathirthrao** [comment]
Thanks for the clarification.
I was able to reproduce the issue and change certainly solves the issue when i start the jtreg test from default display (Without external display launching test from retina/ With external display as default and launching jtreg test from external display). But when i run jtreg test from retina and keep external display as default, test still fails with same behaviour as seen without this fix.

**Comment by mrserb** [comment]
The problem in the multiscreen environment was caused by using the wrong NScreen to calculate the "y" coordinate. The "NSScreen mainScreen" returns the "Returns the screen object containing the window with the keyboard focus" actually the window where the terminal is located, not the screeen(0,0) expected by the test.

**Review by jayathirthrao — APPROVED** [review]

**Pull request closed without merging**